### PR TITLE
Ipv6 privacy extensions

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

File: man/systemd.network.xml [hunk 1]
```diff
@@ -391,6 +391,16 @@
           <literal>no</literal>.</para></listitem>
         </varlistentry>
         <varlistentry>
+          <term><varname>IPv6Privacyxtensions=</varname></term>
```
Comment (zonque) [inline man/systemd.network.xml]: Missing "E"

File: src/network/networkd-network-gperf.gperf [hunk 2]
```diff
@@ -15,69 +15,70 @@ struct ConfigPerfItem;
...
+Network.Gateway,               config_parse_gateway,                           0,                             0
+Network.Domains,               config_parse_domains,                           0,                             offsetof(Network, domains)
+Network.DNS,                   config_parse_strv,                              0,                             offsetof(Network, dns)
+Network.LLMNR,                 config_parse_llmnr,                             0,                             offsetof(Network, llmnr)
+Network.NTP,                   config_parse_strv,                              0,                             offsetof(Network, ntp)
+Network.IPForward,             config_parse_address_family_boolean_with_kernel,0,                             offsetof(Network, ip_forward)
+Network.IPMasquerade,          config_parse_bool,                              0,                             offsetof(Network, ip_masquerade)
+Network.IPv6PrivcyExtensions,  config_parse_int,                               0,                             offsetof(Network, ipv6_privacy_extensions)
```
Comment (zonque) [inline src/network/networkd-network-gperf.gperf]: Typo. Did you actually test this patch set with real config files?

File: man/systemd.network.xml [hunk 3]
```diff
@@ -391,6 +391,16 @@
...
+          <listitem><para>Configures IPv6 Privacy extentions temporary addresses.
+          Preference for Privacy Extensions (RFC3041).
+          <literal>0</literal> disable Privacy Extensions.
+          <literal>1</literal> enable Privacy Extensions, but prefer public addresses over temporary addresses.
+          greater than <literal>1</literal> enable Privacy Extensions and prefer temporary addresses over public addresses.
+          Defaults to
+          <literal>0</literal>.</para></listitem>
```
Comment (zonque) [inline man/systemd.network.xml]: I'm not sure if exposing this strange kernel-level interface in config files is really our best option. I guess a set of well-defined strings would be a better match.

File: src/network/networkd-link.c [hunk 4]
```diff
@@ -1507,6 +1520,28 @@ static int link_set_ipv6_forward(Link *link) {
...
+
+        /* Make this a NOP if IPv6 is not available */
+        if (!socket_ipv6_is_supported())
+                return 0;
+
+        if (!link_ipv6_privacy_extensions_enabled(link))
+                return 0;
+
+        p = strjoina("/proc/sys/net/ipv6/conf/", link->ifname, "/use_tempaddr");
+        snprintf(buf, sizeof(buf), "%u", link->network->ipv6_privacy_extensions);
```
Comment (zonque) [inline src/network/networkd-link.c]: Use `xsprintf()`

**Comment by zonque** [comment]
Nice to see this feature coming. Some comments inline. Also, please provide a proper commit log. This is what goes into git eventually, so it should describe what it does and why.

**Comment by zonque** [comment]
Note that changing the title in GitHub does _not_ suffice as that doesn't alter the actual commit. You have to do amend that in your tree and push a new PR.

File: man/systemd.network.xml (hunk 3 above)
Comment (teg) [inline man/systemd.network.xml]: Yeah, this needs to be an enum, no magic numbers please.

File: src/network/networkd-network-gperf.gperf (hunk 2 above)
Comment (ssahani) [inline src/network/networkd-network-gperf.gperf]: May be I just pushed the wrong version . I just edited private to privacy

File: man/systemd.network.xml (hunk 3 above)
Comment (ssahani) [inline man/systemd.network.xml]: yes I will just modify it

**Pull request closed without merging**